meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -275,7 +276,8 @@ public DirectChildSubscribeResult subscribeDirectChildChange(String key,
 
   @Override
   public boolean subscribeStateChanges(ConnectStateChangeListener listener) {
-    return false;
+    _zkClient.subscribeStateChanges(new StateChangeListenerAdapter(listener));
+    return true;
```

Comment:
If returning true means successful subscription then we should update it's documentation correctly that "it would either return true in case of success or raise an exception" here or in MetaClientInterface.java

Reply:
This is best effort with ZK implementation. We still want to keep the interface behavior.